### PR TITLE
address feedback

### PR DESCRIPTION
**Pull request by taylorn16**

### CODE REVIEW

File: src/Array.ts [hunk 1]
```diff
@@ -44,10 +44,19 @@ const mapi =
...
+ * @remarks
+ * Essentially, this is a map + filter operation where each
+ * element of the array is mapped into an `Option` and an
+ * `isSome` check is used as the filter function.
+ *
  * @example
  * const actual = pipe(
- *     [32, null, 55, undefined, 89] as const,
- *     Array.choose(flow(Option.ofNullish, Option.map(String)))
+ *     [32, null, 55, undefined, 89],   // (number | null | undefined)[]
```
Comment (rmjohnson) [inline src/Array.ts:54]: :chefskiss:

File: src/Array.ts [hunk 2]
```diff
@@ -115,7 +134,7 @@ const tail = <A>(as: readonly A[]): Option<readonly A[]> => {
  * Will return the entire array if `n` is greater
  * than the length of the array.
  *
- * @param count is normalized to a natural number
+ * @param count is normalized to a non-negative integer
```
Comment (rmjohnson) [inline src/Array.ts:137]: :+1:

File: src/Array.ts [hunk 3]
```diff
@@ -289,11 +308,14 @@ const concat =
...
  *
  * @example
+ * // it reads "backwards" when used with `pipe`
  * pipe(
  *     ["a", "b"],
  *     Array.concatFirst(["c", "d"])
  * ); // ["c", "d", "a", "b"]
  *
+ * @example
+ * // it reads better when not used with `pipe`
```
Comment (rmjohnson) [inline src/Array.ts:318]: Good clarification, thanks!

File: src/Array.ts [hunk 4]
```diff
@@ -532,8 +556,11 @@ const except =
         return out
     }
 
-/** Returns the set union of two arrays, defined as the set of elements
+/**
+ * Returns the set union of two arrays, defined as the set of elements
  * contained in both arrays. Remember: sets only contain unique elements.
+ *
+ * If you just need to join two arrays together, use `concat`.
```
Comment (rmjohnson) [inline src/Array.ts:563]: :tada:

File: src/AsyncResult.ts [hunk 5]
```diff
@@ -219,36 +219,7 @@ const match =
...
- * ); // yields "default"
- */
-const matchOrElse =
-    <A, E, R>(matcher: PartialAsyncResultMatcher<A, E, R>) =>
-    (async: AsyncResult<A, E>): Async<R> =>
-    () =>
-        async().then(Result.matchOrElse(matcher))
-
-/** Equivalent to both Async.start and simiply invoking
+/** Equivalent to both Async.start or simply invoking
```
Comment (rmjohnson) [inline src/AsyncResult.ts:222]: Good catch :+1:

File: src/Option.ts [hunk 6]
```diff
@@ -158,6 +115,15 @@ const filter = <A>(f: (a: A) => boolean) =>
  * type guard holds for the wrapped value, returns `Some`
  * with the refined type. `None` is passed through
  * without being checked.
+ *
+ * @example
+ * import { String } from ".."
+ *
+ * pipe(
+ *     Option.Some("cheese" as any),    // Option<any>
+ *     Option.refine(String.isString),  // Option<string> (type is narrowed by the guard)
```
Comment (rmjohnson) [inline src/Option.ts:124]: Great example 👏

File: src/Result.ts [hunk 7]
```diff
@@ -234,9 +193,9 @@ const isErr = <E, A = unknown>(result: Result<A, E>): result is Err<E> =>
 const map2 =
     <A, B, C, E>(map: (a: A, b: B) => C) =>
     (results: readonly [Result<A, E>, Result<B, E>]): Result<C, E> => {
-        if (Result.isOk(results[0]) && Result.isOk(results[1])) {
+        if (isOk(results[0]) && isOk(results[1])) {
```
Comment (rmjohnson) [inline src/Result.ts:196]: :head-tap:

File: src/Result.ts [hunk 8]
```diff
@@ -301,8 +256,9 @@ function tryCatch<A, E = unknown>(
 /** Allows some arbitrary side-effect function to be called
  * using the wrapped Ok value. Useful for trace logging.
  *
- * @param f should not mutate its arguments
- *a
+ * @param f should not mutate its arguments use `map` if you
```
Comment (rmjohnson) [inline src/Result.ts]: :pick: Run-on sentence? At least kinda awkward if not (I'm no english major! =P ) Might just be missing a period?

File: src/function.ts [hunk 9]
```diff
@@ -22,6 +22,8 @@ export const tee =
  * Execute some arbitrary side-effect function using the value
  * resolved from the Promise and return the value unchanged.
  *
+ * If you are working with `Async`s, use `Async.tee` instead.
```
Comment (rmjohnson) [inline src/function.ts:25]: :heart:

**Review by rmjohnson — COMMENTED** [review]
Thanks @taylorn16 !

File: src/Result.ts (hunk 8 above)
Comment (taylorn16) [reply src/Result.ts]: Woops. Totally missed a period there, yep. Good catch!

Comment (taylorn16) [reply src/Result.ts]: [704c7f2](https://github.com/taylorn16/fp-toolkit/pull/19/commits/704c7f24bbfddefbac09a125222c3ed0d3fb9fce)

**Review by rmjohnson — APPROVED** [review]
Great job! :tada:

**Pull request merged**